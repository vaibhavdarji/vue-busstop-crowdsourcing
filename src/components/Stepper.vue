<template>
  <v-stepper v-model="currentStep">
    <v-stepper-header>
      <template v-for="(stepItem, index) in steps">
        <v-stepper-step
          :step="index + 1" :key="index" :complete="currentStep > index + 1">
          {{stepItem.title}}
        </v-stepper-step>
        <v-divider :key="`divider-${index}`"></v-divider>
      </template>
    </v-stepper-header>
    <v-stepper-items>
      <template v-for="(stepItem, index) in steps">
        <v-stepper-content :step="index + 1" :key="index">
          <component :is="stepItem.component" @onPreviousStep="onPrevious" @onNextStep="onNext"></component>
        </v-stepper-content>
      </template>
    </v-stepper-items>
  </v-stepper>
</template>

<script>
import Donate from '@/components/Donate.vue';
import UserInfo from '@/components/UserInfo.vue';
import Payment from '@/components/Payment.vue';
import CompleteTransaction from '@/components/CompleteTransaction.vue';
export default {
  name: 'Stepper',
  data () {
    return {
      currentStep: 1,
      steps: [{
        title: "Donate",
        component: Donate,
        ref: "donate"
      }, {
        title: "Your Information",
        component: UserInfo,
        ref: 'personalInfo'
      },
      {
        title: "Payment Information",
        component: Payment,
        ref: 'paymentInfo'
      }, {
        title: "Payment Successful",
        component: CompleteTransaction,
        ref: 'completeInfo'
      }]
    };
  },

  components: {
    Donate,
    UserInfo,
    Payment
  },

  methods: {
    onNext () {
      if (this.currentStep === this.steps.length) {
        this.$router.push('/');
      } else {
        this.currentStep = this.currentStep + 1;
      }
    },

    onPrevious () {
      
      this.currentStep = this.currentStep - 1;
    }
  }
}
</script>

