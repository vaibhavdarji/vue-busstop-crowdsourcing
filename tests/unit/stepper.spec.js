import { shallowMount } from "@vue/test-utils";
import Stepper from "@/components/Stepper.vue";
import Vue from "vue";
import Vuetify from "vuetify";

Vue.use(Vuetify);

describe("Stepper.vue", () => {
  it("should match snapshot", () => {
    const wrapper = shallowMount(Stepper, {
      Vue
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});