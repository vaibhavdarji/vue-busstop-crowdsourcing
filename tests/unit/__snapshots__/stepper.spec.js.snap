// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper.vue should match snapshot 1`] = `
<v-stepper-stub
  value="1"
>
  <v-stepper-header-stub>
    <v-stepper-step-stub
      color="primary"
      completeicon="$vuetify.icons.complete"
      editicon="$vuetify.icons.edit"
      erroricon="$vuetify.icons.error"
      rules=""
      step="1"
    >
      
        Donate
      
    </v-stepper-step-stub>
     
    <v-divider-stub />
    <v-stepper-step-stub
      color="primary"
      completeicon="$vuetify.icons.complete"
      editicon="$vuetify.icons.edit"
      erroricon="$vuetify.icons.error"
      rules=""
      step="2"
    >
      
        Your Information
      
    </v-stepper-step-stub>
     
    <v-divider-stub />
    <v-stepper-step-stub
      color="primary"
      completeicon="$vuetify.icons.complete"
      editicon="$vuetify.icons.edit"
      erroricon="$vuetify.icons.error"
      rules=""
      step="3"
    >
      
        Payment Information
      
    </v-stepper-step-stub>
     
    <v-divider-stub />
    <v-stepper-step-stub
      color="primary"
      completeicon="$vuetify.icons.complete"
      editicon="$vuetify.icons.edit"
      erroricon="$vuetify.icons.error"
      rules=""
      step="4"
    >
      
        Payment Successful
      
    </v-stepper-step-stub>
     
    <v-divider-stub />
  </v-stepper-header-stub>
   
  <v-stepper-items-stub>
    <v-stepper-content-stub
      step="1"
    >
      <donate-stub />
    </v-stepper-content-stub>
    <v-stepper-content-stub
      step="2"
    >
      <userinfo-stub />
    </v-stepper-content-stub>
    <v-stepper-content-stub
      step="3"
    >
      <payment-stub />
    </v-stepper-content-stub>
    <v-stepper-content-stub
      step="4"
    >
      <completetransaction-stub />
    </v-stepper-content-stub>
  </v-stepper-items-stub>
</v-stepper-stub>
`;
